libethereum/Client.cpp

```diff
@@ -851,7 +851,7 @@ pair<h256, Address> Client::submitTransaction(TransactionSkeleton const& _t, Sec
     if (ts.gasPrice == Invalid256)
         ts.gasPrice = gasBidPrice();
     if (ts.gas == Invalid256)
-        ts.gas = min<u256>(gasLimitRemaining() / 5, balanceAt(ts.from) / ts.gasPrice);
+        ts.gas = min<u256>(gasLimitRemaining() / 5, ts.gasPrice > 0 ? balanceAt(ts.from) / ts.gasPrice : balanceAt(ts.from) / gasBidPrice());
```

Comment:
Why do you use the big price here?

Reply:
The bid price is used as a default value when one doesn't explicitly supply a gas price:
https://github.com/ethereum/cpp-ethereum/blob/8d1c17ba324455c228d90a1997f281be8266adc9/libethereum/Client.cpp#L851-L852
As a result, I thought it was a suitable value to fall back to when computing the gas limit. I'm of course open to suggestions for a better value to use. 

An alternate approach could be to replace the gas computation entirely with a default gas constant (I think geth and testrpc use 90,000) - in practice this would probably result in lower gas limits being used most of the time and cover value transfer and many contract creation txs (which have base gas costs of 21K and 32K, respectively). What do you think?

Reply:
You bring up a good point - I actually chatted with @gumb0  about this last week (see #3121), I think that we should allow transactions with gasPrice = 0 since this appears to be allowed by the protocol (and geth accepts them). To address this we can remover the gas computation entirely and fall back to a default gas limit (I'm in favor of using geth/testrpc's 90,000). What do you think?

Reply:
Ok, I'm for using 90000 constant. Just expose the constant somewhere by name "defaultTransactionGas". This calculation might be more confusing than useful.

Reply:
Agreed, the computation is confusing (especially to someone who isn't intimately familiar with the codebase) and tends to result in very high gas limits (unless you have an extremely small account balance).